### PR TITLE
Feature: Add and consume trade role and status in deal history

### PR DESCRIPTION
**Pull request by TheCharlatan**
When listing deals the deal status and trade role are now always shown. This data is also available through the grpc server.

### CODE REVIEW

**Review by h4sh3d — COMMENTED** [review]
I'm testing this while building the GUI, LGTM but we need the `Revoke` status

File: tests/docker-compose.yml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-version: "3.8"
+version: "3.6"
```
[the remark this answers is not in the document]
Comment (TheCharlatan) [reply tests/docker-compose.yml:1]: Oh, whoops

File: src/grpcd/proto/farcaster.proto [hunk 2]
```diff
@@ -110,7 +118,15 @@ message DealInfo {
     Blockchain accordant_blockchain = 10;
     string node_id = 11;
     string peer_address = 12;
-    string encoded_deal = 13;
+}
+
+enum DealStatus {
```
[the remark this answers is not in the document]
Comment (TheCharlatan) [reply src/grpcd/proto/farcaster.proto:123]: There is no such thing as a revoke status currently in the node DealStatus, I'll add it in a separate commit.

[the remark this answers is not in the document]
Comment (TheCharlatan) [reply src/grpcd/proto/farcaster.proto:123]: [1906656](https://github.com/farcaster-project/farcaster-node/pull/861/commits/1906656e0784106d3427c097730f457f96ececc0)

**Review by h4sh3d — APPROVED** [review]
tACK 9116501a88e3936fb4a65d7b0fbe247cc036087f

**Pull request merged**